### PR TITLE
feat(tagsInput): Add global character limit option

### PR DESCRIPTION
**Pull request by iures**
global max lenght blocks user from typing after the limit
has reached. The limit takes into account existing tags, prefix
and text in the input

### CODE REVIEW

File: build/ng-tags-input.js [hunk 1]
```diff
@@ -201,6 +201,33 @@ tagsInput.directive('tagsInput', ["$timeout","$document","tagsInputConfig", func
...
+
+        self.getCharacterLength = function() {
+            var allItems = self.getAllItems();
+            if(allItems.length === 0) {
+                return 0;
+            }
+
+            var length = self.getAllItems().map(function(item) {
+                return item.text;
+            }).reduce(function(previousValue, currentValue, index, array) {
```
Comment (kevinc) [inline build/ng-tags-input.js]: dat functional programming though

File: build/ng-tags-input.js [hunk 2]
```diff
@@ -201,6 +201,33 @@ tagsInput.directive('tagsInput', ["$timeout","$document","tagsInputConfig", func
...
+                return (previousValue + currentValue);
+            }).length;
```
Comment (kevinc) [inline build/ng-tags-input.js]: why you need the length?

**Comment by tjhosford** [comment]
lgtm, nice

File: build/ng-tags-input.js (hunk 2 above)
Comment (tjhosford) [inline build/ng-tags-input.js]: Yea this confused me at first.  It's because reduce returns a string

Comment (kevinc) [inline build/ng-tags-input.js]: ah yea...thought we were mapping lengths

Comment (tjhosford) [inline build/ng-tags-input.js]: This might be a bit more readable

`self.getAllItems().map(function(item) {return item.text;}).join("").length`

Comment (kevinc) [inline build/ng-tags-input.js]: join on " " instead to take spaces into account though

Comment (iures) [inline build/ng-tags-input.js]: It returns the concatenated texts. Not sure if thats a word =)

File: build/ng-tags-input.js (hunk 1 above)
Comment (iures) [inline build/ng-tags-input.js]: right? Any suggestions? I'm itching to change that

File: src/tags-input.js [hunk 3]
```diff
@@ -249,6 +268,22 @@ tagsInput.directive('tagsInput', function($timeout, $document, tagsInputConfig)
                 events.trigger('input-change', scope.newTag.text);
```
Comment (kevinc) [inline src/tags-input.js:268]: what's this event?

Comment (iures) [inline src/tags-input.js:268]: input-change? This was already there, I didn't add it. This seems to be related to when a new-tag is created and the input needs to be reset

**Comment by tjhosford** [comment]
lgtm

**Pull request closed without merging**